ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/ChainGenerator/Tests/BitVector.hs

```diff
@@ -141,3 +158,56 @@ prop_findIthZeroInV testSetup = case testSetup of
                       $ QC.counterexample "There are not testIndex-many post-polarization-0s preceding i!"
                       $ targetsInPrecedingWords QC.=== C.getCount testIndex
                   )
+
+
+data FillInWindowSetup =
+    FillInWindowSetup
+      SomePol
+      Int -- ^ k
+      Int -- ^ size of the window to fill
+      QCGen -- ^ random generator
+      (V.Vector S) -- ^ the vector to fill
+  deriving (Show)
+
+instance QC.Arbitrary FillInWindowSetup where
+  arbitrary = do
+      pol <- QC.arbitrary
+      QC.NonNegative k <- QC.arbitrary
+      QC.NonNegative k1 <- QC.arbitrary
+      QC.NonNegative k2 <- QC.arbitrary
+      qcGen <- QC.arbitrary
+      let szV = k1 + 1
+          szW = max k szV + k2 -- k <= szW && szV <= szW
+      ss <- QC.vectorOf szV QC.arbitrary
+      pure $ FillInWindowSetup pol k szW qcGen (V.fromList ss)
+
+prop_fillInWindow :: FillInWindowSetup -> QC.Property
+prop_fillInWindow
+        (FillInWindowSetup
+          (SomePol pol)
+          k
+          szW
+          qcGen
+          v
+        ) = R.runSTGen_ qcGen $ \g -> do
+    mv <- C.unsafeThawV $ C.Vector v
```

Comment:
`fillInWindow` is writing to `mv`, so using [`unsafeThawV`](https://hackage.haskell.org/package/vector-0.13.1.0/docs/Data-Vector.html#v:unsafeThaw) is indeed unsafe here:

> All in all, attempts to modify a vector produced by unsafeThaw fall out of domain of software engineering and into realm of black magic, dark rituals, and unspeakable horrors.

(Existing usages `unsafeThawV` are read-only.)

Reply:
Changed to use `V.thaw` insted.